extension/src/pipeline/util.ts

```diff
@@ -85,13 +87,45 @@ export type PlotConfigData = {
   y: string
 }
 
+const getFieldOptions = (data: unknown): string[] => {
+  const isArray = Array.isArray(data)
+  const isObj = isObject(data)
+  if (!isArray && !isObj) {
+    return []
+  }
+
+  const maybeFieldsObjArr = isArray ? data : data[Object.keys(data)[0]]
+
+  if (!Array.isArray(maybeFieldsObjArr)) {
+    return []
+  }
+
+  const maybeFieldsObj: unknown = maybeFieldsObjArr[0]
+  return isObject(maybeFieldsObj) ? Object.keys(maybeFieldsObj) : []
+}
+
 export const pickPlotConfiguration = async (): Promise<
   PlotConfigData | undefined
 > => {
-  // TBD data file validation will be in next pr
-  const file = (await pickDataFile()) as string
-  const data = (await parseDataFile(file)) as Record<string, unknown>[]
-  const keys = Object.keys(data[0])
+  const file = await pickDataFile()
+
+  if (!file) {
+    return
+  }
+
+  const data = await parseDataFile(file)
+  const failedToParseMessage =
+    'Failed to find field options for plot data. Is your file following DVC plot guidelines for [JSON/YAML](https://dvc.org/doc/command-reference/plots/show#example-hierarchical-data) or [CSV/TSV](https://dvc.org/doc/command-reference/plots/show#example-tabular-data) files?'
```

Comment:
[I] I would split this and show a different message when parsing throws an error/returns nothing vs the data has < 2 keys. On first read `field options for plot data` is ambiguous/needs more context. 

As an end goal we would want to tell the user:

1. If the file did not parse successfully
2. If the file was empty
3. There are not enough keys to generate a plot

As splitting 1/2 out would require a large refactor I'd suggest breaking out the message to 1+2 or 3.

E.g. 

`'Failed to parse the requested file. Does the file contain data and follow the DVC plot guidelines for [JSON/YAML](https://dvc.org/doc/command-reference/plots/show#example-hierarchical-data) or [CSV/TSV](https://dvc.org/doc/command-reference/plots/show#example-tabular-data) files?'`
and 
`'The request file does not contain enough keys (columns) to generate a plot. Does the file follow the DVC plot guidelines for [JSON/YAML](https://dvc.org/doc/command-reference/plots/show#example-hierarchical-data) or [CSV/TSV](https://dvc.org/doc/command-reference/plots/show#example-tabular-data) files?'`

Reply:
Good idea! Update the messaging!